tilecloud_chain/generate.py

```diff
@@ -141,24 +141,37 @@ def wrong_content_type_to_error(tile):
                     if tile.content_type.find("application/vnd.ogc.se_xml") == 0:
                         tile.error = "WMS server error: {}".format((
                             self._re_rm_xml_tag.sub(
-                                '', tile.data.decode('utf-8') if PY3 else tile.data
+                                '', tile.error.decode('utf-8') if PY3 else tile.error
                             )
                         ))
                     else:  # pragma: no cover
                         tile.error = "{} is not an image format, error: {}".format(
                             tile.content_type,
-                            tile.data
+                            tile.error
                         )
                 return tile
             gene.imap(wrong_content_type_to_error)
+            gene.add_error_filters()
 
             if options.role == 'hash':
-                if 'meta' in gene.layer:
+                if gene.layer.get('meta', False):
                     gene.imap(HashLogger('empty_metatile_detection'))
+            elif options.role == 'slave':
+                droppers = {}
+                for lname, layer in gene.layers.items():
+                    if 'empty_metatile_detection' in layer:
+                        empty_tile = layer['empty_metatile_detection']
+                        droppers[lname] = HashDropper(
+                            empty_tile['size'], empty_tile['hash'], store=self.cache_tilestore,
+                            queue_store=self.sqs_tilestore,
+                            count=self.count_metatiles_dropped,
+                        )
+                if droppers:
+                    gene.imap(MultiHashDropper(droppers))
             elif not options.near:
                 # Discard tiles with certain content
-                if 'empty_tile_detection' in gene.layer:
-                    empty_tile = gene.layer['empty_tile_detection']
+                if gene.layer and 'empty_metatile_detection' in gene.layer:
+                    empty_tile = gene.layer['empty_metatile_detection']
```

Comment:
Because a typo in 49f5612dfbc69f12cf6d036ff27801a2483ba231 changed it to tile. This block was always meant to be the metatile dropper